src/index.ts

```diff
@@ -148,13 +161,21 @@ function path(template: string, params: ParamMap) {
  * ```
  */
 export function join(part1: string, separator: string, part2: string) {
-  const p1 = part1.endsWith(separator) ? part1.slice(0, -separator.length) : part1;
-  const p2 = part2.startsWith(separator) ? part2.slice(separator.length) : part2;
-  return p1 === '' || p2 === ''
-    ? p1 + p2
-    : p1 + separator + p2;
+  const p1 = part1.endsWith(separator)
+    ? part1.slice(0, -separator.length)
+    : part1;
+  const p2 = part2.startsWith(separator)
+    ? part2.slice(separator.length)
+    : part2;
+  return p1 === "" || p2 === "" ? p1 + p2 : p1 + separator + p2;
```

Comment:
I see why you reformatted the first two lines - they are slightly easier to read this way - but I see no reason to put the return statement on a single line. It's become very hard to read. Could you please format it again?